pandas/io/parsers.py

```diff
@@ -1421,6 +1422,54 @@ def __init__(self, kwds):
         # keep references to file handles opened by the parser itself
         self.handles = []
 
+    def _validate_parse_dates_presence(self, columns: List[str]) -> None:
+        """
+        Check if parse_dates are in columns.
+
+        If user has provided names for parse_dates, check if those columns
+        are available.
+
+        Parameters
+        ----------
+        columns : list
+            List of names of the dataframe.
+
+        Raises
+        ------
+        ValueError
+            If column to parse_date is not in dataframe.
+
+        """
+        cols_needed: Iterable
+        if isinstance(self.parse_dates, list):
```

Comment:
use `is_list_like(self.parse_dates)` here (e.g. this will not work with a tuple, ndarray etc; this will exclude a dict (or you can simply put the dict check first if you want)